prov/efa/test/efa_unit_test_mocks.c

```diff
@@ -5,68 +5,128 @@
 #include <setjmp.h>
 #include <cmocka.h>
 #include "efa.h"
+#include "efa_unit_test_mocks.h"
 
-void __real_rxr_pkt_handle_send_completion(struct rxr_ep *ep, struct rxr_pkt_entry *pkt_entry);
+/* mock of rdma-core functions */
+uint8_t g_ibv_gid_mock[EFA_GID_LEN] = {0xfe, 0x80, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x10, 0xcb, 0x06, 0xff, 0xfe, 0x66, 0x83, 0x2f};
 
-void __wrap_rxr_pkt_handle_send_completion(struct rxr_ep *ep, struct rxr_pkt_entry *pkt_entry)
+int g_ibv_create_ah_call_counter = 0;
+struct ibv_ah *efa_mock_ibv_create_ah_increase_call_counter(struct ibv_pd *pd, struct ibv_ah_attr *attr)
 {
-	check_expected(ep);
-	check_expected(pkt_entry);
+	g_ibv_create_ah_call_counter += 1;
+	return  __real_ibv_create_ah(pd, attr);
+}
 
-	if ((int)mock() == 4242) {
-		__real_rxr_pkt_handle_send_completion(ep, pkt_entry);
-	}
+int efa_mock_efadv_query_device_return_mock(struct ibv_context *ibv_ctx,
+					    struct efadv_device_attr *attr,
+					    uint32_t inlen)
+{
+	return mock();
 }
 
-void __real_rxr_pkt_handle_recv_completion(struct rxr_ep *ep,
-										   struct rxr_pkt_entry *pkt_entry,
-										   enum rxr_lower_ep_type lower_ep_type);
+struct ibv_send_wr *g_ibv_send_wr_list_head;
+struct ibv_send_wr *g_ibv_send_wr_list_tail;
```

Comment:
Global vars are convenient but it could lead to undetermined state while running many tests.

When should we reset `g_ibv_send_wr_list_head/tail`? A comment would help.

Reply:
added a comment.

Also added code to verify memory is released.